src/cache.rs

```diff
@@ -133,6 +137,52 @@ impl BlockchainDbMeta {
 
         Self { cfg_env: env.cfg.clone(), block_env: env.block, hosts: BTreeSet::from([host]) }
     }
+
+    /// Sets the chain_id in the [CfgEnv] of this instance.
+    ///
+    /// Remaining fields of [CfgEnv] are left unchanged.
+    pub const fn with_chain_id(mut self, chain_id: u64) -> Self {
+        self.cfg_env.chain_id = chain_id;
+        self
+    }
+
+    /// Sets the [BlockEnv] of this instance using the provided [alloy_rpc_types::Block]
+    pub fn with_block<T: TransactionResponse>(mut self, block: alloy_rpc_types::Block<T>) -> Self {
+        self.block_env = BlockEnv {
+            number: U256::from(block.header.number()),
+            coinbase: block.header.coinbase(),
+            timestamp: U256::from(block.header.timestamp()),
+            difficulty: U256::from(block.header.difficulty()),
+            basefee: block.header.base_fee_per_gas().map(U256::from).unwrap_or_default(),
+            gas_limit: U256::from(block.header.gas_limit()),
+            prevrandao: block.header.mix_hash(),
+            blob_excess_gas_and_price: Some(BlobExcessGasAndPrice::new(
+                block.header.excess_blob_gas.map(|gas| gas as u64).unwrap_or_default(),
+            )),
+        };
+
+        self
+    }
+
+    /// Infers the host from the provided url and adds it to the set of hosts
+    pub fn with_url(mut self, url: String) -> Self {
```

Comment:
this should accept &str
```suggestion
    pub fn with_url(mut self, url: &str) -> Self {
```